crates/ironrdp-server/src/server.rs

```diff
@@ -385,7 +385,9 @@ impl RdpServer {
                             break;
                         }
                         ServerEvent::GetLocalAddr(tx) => {
-                            let _ = tx.send(self.local_addr);
+                            if tx.send(self.local_addr).is_err() {
+                                bail!("failed to send the local address");
+                            }
```

Comment:
Greatly changing the behavior: we are short-circuiting the code when an error happens, but that’s not how the code is supposed to flow. Maybe it’s fine, maybe its not; there are several similar places where we should be careful.